apps/api/src/lib/db/migrations/0041_location_metas.sql

```
@@ -0,0 +1,105 @@
+CREATE TABLE "map_group_location_metas" (
+	"location_id" bigint NOT NULL,
+	"meta_id" integer NOT NULL,
+	"map_group_id" integer NOT NULL,
+	CONSTRAINT "map_group_location_metas_location_id_meta_id_pk" PRIMARY KEY("location_id","meta_id")
+);
+--> statement-breakpoint
+ALTER TABLE "map_group_locations" ALTER COLUMN "extra_tag" DROP NOT NULL;--> statement-breakpoint
+CREATE UNIQUE INDEX "map_group_locations_id_map_group_unique" ON "map_group_locations" USING btree ("id","map_group_id");--> statement-breakpoint
+CREATE UNIQUE INDEX "metas_id_map_group_unique" ON "metas" USING btree ("id","map_group_id");--> statement-breakpoint
+ALTER TABLE "map_group_location_metas" ADD CONSTRAINT "map_group_location_metas_location_group_fk" FOREIGN KEY ("location_id","map_group_id") REFERENCES "public"."map_group_locations"("id","map_group_id") ON DELETE cascade ON UPDATE no action;--> statement-breakpoint
```

SQLFluff (4.2.2)

[error] ADD CONSTRAINT ... FOREIGN KEY should use NOT VALID to avoid locking the table while validating existing rows.

(PG01)